### PR TITLE
Document supported Go versions

### PR DESCRIPTION
**Pull request by andrewkroh**
Establish a policy of supporting the two most recent Go versions.

### CODE REVIEW

**Review by kaanyalti — APPROVED** [review]

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       matrix:
         go:
           - 1.22.x
-          - 1.18.x
+          - 1.21.9
```
Comment (efd6) [inline .github/workflows/go.yml:37]:
```suggestion
          - 1.21.x
```
?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -99,8 +99,8 @@ jobs:
     runs-on: ubuntu-latest
     env:
       # Only GITHUB_* are passed into the VM.
-      GITHUB_GOLANG_VERSION: 1.22.1
-    name: test (1.22.1, freebsd-14.0, cgo)
+      GITHUB_GOLANG_VERSION: 1.22.2
+    name: test (1.22.2, freebsd-14.0, cgo)
```
[the remark this answers is not in the document]
Comment (andrewkroh) [reply .github/workflows/go.yml:103]: This one uses GVM. GVM does not support the 1.22.x format. Need to speak to the author about that.

[the remark this answers is not in the document]
Comment (efd6) [reply .github/workflows/go.yml:103]: I figured that might be the case.

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -53,10 +53,10 @@ jobs:
...
           - {cgo: cgo, os: ubuntu-22.04}
           # Limit the OS variants tested with the earliest supported Go version (save resources).
-          - {go: 1.18.10, os: macos-11}
-          - {go: 1.18.10, os: macos-12}
-          - {go: 1.18.10, os: windows-2019}
-          - {go: 1.18.10, os: ubuntu-22.04}
+          - {go: 1.21.9, os: macos-11}
+          - {go: 1.21.9, os: macos-12}
+          - {go: 1.21.9, os: windows-2019}
+          - {go: 1.21.9, os: ubuntu-22.04}
```
[the remark this answers is not in the document]
Comment (andrewkroh) [reply .github/workflows/go.yml:59]: IIRC, the matrix variable gets expanded to the actual version so excluding with 1.21.x doesn't match. This is why I switched away from "x" here. But I'll try this again and see, it has been a long time.

[the remark this answers is not in the document]
Comment (andrewkroh) [reply .github/workflows/go.yml]: The exclusions are working as expected now. Changing to 1.21.x. Thanks 👍

**Review by efd6 — APPROVED** [review]

**Pull request merged**